### PR TITLE
initial group processor

### PR DESCRIPTION
**Pull request by chdietz**

### CODE REVIEW

**Comment by chdietz** [comment]
init group processor

File: group-processor/src/main/java/org/opentestsystem/rdw/ingest/group/GroupProcessorApplication.java [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+import org.springframework.boot.autoconfigure.SpringBootApplication;
+import org.springframework.cache.annotation.EnableCaching;
+import org.springframework.context.annotation.Import;
+import org.springframework.context.annotation.PropertySource;
+
+/**
+ * App for Group Processor
+ */
+@SpringBootApplication
+@EnableCaching
```
Comment (agorina) [inline group-processor/src/main/java/org/opentestsystem/rdw/ingest/group/GroupProcessorApplication.java]: Do you need this?

File: group-processor/build.gradle [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+    compile 'org.opentestsystem.rdw.common:rdw-common-status'
+    compile 'org.opentestsystem.rdw.common:rdw-common-utils'
+    compile 'mysql:mysql-connector-java'
+    compile "org.springframework.boot:spring-boot-devtools"
+
+    compile 'org.apache.commons:commons-csv:+'
+
+    compile project(':rdw-ingest-common')
+    testCompile project(path: ':rdw-ingest-common', configuration: 'tests')
+}
```
Comment (agorina) [inline group-processor/build.gradle]: I think you also need to add this new module to settings.gradle.

File: group-processor/src/main/java/org/opentestsystem/rdw/ingest/group/GroupProcessorConfiguration.java [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+import org.springframework.cloud.stream.messaging.Sink;
+import org.springframework.integration.annotation.ServiceActivator;
+import org.springframework.messaging.Message;
+
+
+@EnableBinding(Sink.class)
+public class GroupProcessorConfiguration {
+    private static final Logger logger = LoggerFactory.getLogger(GroupProcessorConfiguration.class);
+
+    private final GroupProcessor examMigrator;
```
Comment (agorina) [inline group-processor/src/main/java/org/opentestsystem/rdw/ingest/group/GroupProcessorConfiguration.java]: examMigrator is  a left over from copy&paste.

File: group-processor/src/main/java/org/opentestsystem/rdw/ingest/group/GroupProcessorConfiguration.java [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+
+    @Autowired
+    public GroupProcessorConfiguration(final GroupProcessor examMigrator) {
+        this.examMigrator = examMigrator;
+    }
+
+    @ServiceActivator(inputChannel = Processor.INPUT)
+    public void process(final Message<?> message) {
+
+        // todo: need to define message for migrate.  Need Jira for this.  Start with generation in exam-processor?
```
Comment (agorina) [inline group-processor/src/main/java/org/opentestsystem/rdw/ingest/group/GroupProcessorConfiguration.java]: And this todo seems like a leftover from c&p

File: group-processor/src/main/java/org/opentestsystem/rdw/ingest/group/service/GroupProcessor.java [hunk 5]
```diff
@@ -0,0 +1,13 @@
+package org.opentestsystem.rdw.ingest.group.service;
+
+
+/**
+ * An interface for migrating an exam
+ */
+public interface GroupProcessor {
+
+    /**
+     * @param examRecord the record to process
```
Comment (agorina) [inline group-processor/src/main/java/org/opentestsystem/rdw/ingest/group/service/GroupProcessor.java]: examRecord again.

File: group-processor/src/main/java/org/opentestsystem/rdw/ingest/group/service/impl/DefaultGroupProcessor.java [hunk 6]
```diff
@@ -0,0 +1,31 @@
...
+     *
+     * @param message A migrate message sent after successful data warehouse ingest
+     */
+    @Override
+    public void process(final String message) {
+
+        // todo: Need to design message to be created post exam processor, need Jira for this.
+        logger.info("Received migrate message: " + (message.length() > 80 ? message.substring(0, 80) + "..." : message));
+    }
+
```
Comment (agorina) [inline group-processor/src/main/java/org/opentestsystem/rdw/ingest/group/service/impl/DefaultGroupProcessor.java]: All comments are outdated here.

File: group-processor/src/main/java/org/opentestsystem/rdw/ingest/group/GroupProcessorApplication.java (hunk 1 above)
Comment (chdietz) [inline group-processor/src/main/java/org/opentestsystem/rdw/ingest/group/GroupProcessorApplication.java]: do not need cache, will remove, add it when we need it

File: settings.gradle [hunk 7]
```diff
@@ -4,6 +4,7 @@ include 'exam-processor'
 include 'package-processor'
 include 'migrate-reporting'
 include 'task-service'
+include 'group-processor'
```
Comment (chdietz) [inline settings.gradle]: forgot to push this

File: group-processor/build.gradle (hunk 2 above)
Comment (chdietz) [inline group-processor/build.gradle]: did forgot to push it

File: group-processor/src/main/java/org/opentestsystem/rdw/ingest/group/GroupProcessorConfiguration.java (hunk 3 above)
Comment (chdietz) [inline group-processor/src/main/java/org/opentestsystem/rdw/ingest/group/GroupProcessorConfiguration.java]: good catch changed

File: group-processor/src/main/java/org/opentestsystem/rdw/ingest/group/service/GroupProcessor.java (hunk 5 above)
Comment (chdietz) [inline group-processor/src/main/java/org/opentestsystem/rdw/ingest/group/service/GroupProcessor.java]: good catch changed

File: group-processor/src/main/java/org/opentestsystem/rdw/ingest/group/service/impl/DefaultGroupProcessor.java (hunk 6 above)
Comment (chdietz) [inline group-processor/src/main/java/org/opentestsystem/rdw/ingest/group/service/impl/DefaultGroupProcessor.java]: good catch changed

File: group-processor/src/main/java/org/opentestsystem/rdw/ingest/group/service/impl/DefaultGroupProcessor.java [hunk 8]
```diff
@@ -0,0 +1,31 @@
...
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+
+@Service
+public class DefaultGroupProcessor implements GroupProcessor {
+    private static final Logger logger = LoggerFactory.getLogger(DefaultGroupProcessor.class);
+
+    @Autowired
```
Comment (malaffoon) [inline group-processor/src/main/java/org/opentestsystem/rdw/ingest/group/service/impl/DefaultGroupProcessor.java]: this didn't give you a warning? autowiring with no params

**Pull request merged**